### ISSUE TITLE
[Request] Configs for Living Armor Levels

### ISSUE THREAD

**Issue by Rosethorns** [WayofTime/BloodMagic#1110]
I would like to be able to tweak what my players can achieve with armor upgrade levels.

A living example of this is currently with 300 upgrade points a player can get up to healthy level 9 which gives them 22 extra hearts for a total of 32. Personally I find this way too powerful for my packs designs. And would like to ensure that at most by level 9 they can have an extra 10 hearts.

Of course situations like this can apply to a great deal of the other abilities as well So I think all round it would be great if we could set level reward levels ourselves.

TL;DR Please think about adding configs for living armor levels.

**Comment by MrNinja93** [comment]
To add maybe having a user set max level for each upgrade could be a decent way to attain this.

e.g

Healthy [Range 1-10, Default 10]
"x"
Body Builder [Range 1-5, Default 5]
"x" 

etc etc..

**Comment by Rosethorns** [comment]
I disagree. because then you are lowering overall levels to aquire. What if you want 10 levels for healthy overall. experience curve and everything?
If you simply restrict the range you make the maximum quick to achieve.

**Comment by MrNinja93** [comment]
I suppose hypothetically you could combine the two, and behind the scenes shift that experience curve/upgrade cost per level based on max level entered. If max is 10 and someone's entered 5 make 5 as hard as 10 to get/double the "exp" needed. something along those lines.

**Comment by Rosethorns** [comment]
That would be better, you could literally make the entries curves too. rather than individual level control
max level: 10
max health: 20
max points: 320

then math the fek out of it.

**Comment by TehNut** [comment]
I'll need to find some time to implement this. It'll require rewriting a good chunk of the upgrade system. Don't expect it anytime soon. Sorry.

**Comment by Rosethorns** [comment]
Hey, just as long as its something to look forward to, I'm happy xD

**Comment by wrincewind** [comment]
This is resolved in 1.21+ (or, will be on release), but fixing it in 1.20 would involve significant rewrites. Technically any user can do this by editing the jar as the values are in json files, but that's a lot to ask of the end-user.
